pyecsca/sca/stacked_traces/stacked_traces.py

```diff
@@ -0,0 +1,206 @@
+from numba import cuda
+import numpy as np
+from public import public
+from typing import Any, Mapping, MutableSequence, Tuple, Union
+from math import sqrt
+
+from pyecsca.sca.trace.trace import CombinedTrace
+
+
+@public
+class StackedTraces:
+    """Samples of multiple traces and metadata"""
+
+    meta: Mapping[str, Any]
+    samples: np.ndarray
+
+    def __init__(
+            self, samples: np.ndarray,
+            meta: Mapping[str, Any] = None) -> None:
+        if meta is None:
+            meta = {}
+        self.meta = meta
+        self.samples = samples
+
+    @classmethod
+    def fromarray(cls, traces: MutableSequence[np.ndarray],
+                  meta: Mapping[str, Any] = None) -> 'StackedTraces':
+        min_samples = min(map(len, traces))
+        for i, t in enumerate(traces):
+            traces[i] = t[:min_samples]
```

Comment:
I'm not sure we can do this as it modifies the originally submitted traces. Is this done somewhere else in pyecsca?